### PR TITLE
Add builtins folder and JS Date data

### PR DESCRIPTION
**Pull request by Elchi3**
- Moved builtins into a builtins folder (e.g. `function` is a builtin object, a statement, and an expression, so there would be conflicts if things would be all under the same folder)
- Added Date data https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date
- Renamed to uppercase `Promise` namespace to be consistent with what we have in API data.

### CODE REVIEW

File: javascript/builtins/Date.json [hunk 1]
```diff
@@ -0,0 +1,3548 @@
...
+                    "version_added": true
+                  },
+                  "ie": {
+                    "version_added": true
+                  },
+                  "ie_mobile": {
+                    "version_added": true
+                  },
+                  "nodejs": {
+
```
Comment (wbamberg) [inline javascript/builtins/Date.json]: I think this is given as "9" in the wiki: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString#Browser_compatibility

Comment (wbamberg) [inline javascript/builtins/Date.json]: Values here are different to values in the wiki (this has more falses and fewer nulls).

Comment (wbamberg) [inline javascript/builtins/Date.json]: This has falses where the wiki has nulls.

Comment (wbamberg) [inline javascript/builtins/Date.json]: This is missing "generic method", compared with the wiki.

**Comment by Elchi3** [comment]
> A few bits of this API are marked as deprecated, but there's no indication of that here. Should they be marked "obsolete"? Should we rename "obsolete" to "deprecated"? 

I thought about this and I think the answer is: yes, we should rename to deprecated. See #265.

File: javascript/builtins/Date.json (hunk 1 above)
Comment (Elchi3) [inline javascript/builtins/Date.json]: tbh, it's unclear to me what this means. I'm not sure it's worth to mention. It's not explained in the doc what it means.

**Comment by Elchi3** [comment]
> Also these are not quite alphabetized, should they be?

Sorted them. Thanks!

**Comment by Elchi3** [comment]
> A few bits of this API are marked as deprecated, but there's no indication of that here. Should they be marked "obsolete"? Should we rename "obsolete" to "deprecated"?

I marked them as `obsolete: true` for now. I assume we'll rename obsolete to deprecated.

> I checked this against the Wiki pages, and found a few minor discrepancies, which I've noted. I don't know whether the Wiki or this is better.

Sometimes I added what I think makes more sense (e.g. same engine under the hood, so it should be "true" instead of "null" or "false"). I think the MDN data is not perfect, and I hope that we'll collectively correct this data with the community help in this repo. Fact-checking while migrating initial data sets slows us down. I don't know if "best guess" is good for migrating or strictly migrating what's currently in the wiki.

**Pull request merged**